memory-db/src/lib.rs

```diff
@@ -421,53 +460,30 @@ where
 	}
 }
 
-// `no_std` implementation requires that hasmap
-// is implementated in parity-util-mem, that
-// is currently not the case.
-#[cfg(feature = "std")]
-impl<H, KF, T> MallocSizeOf for MemoryDB<H, KF, T>
+impl<H, KF, T, M> MallocSizeOf for MemoryDB<H, KF, T, M>
 where
 	H: KeyHasher,
 	H::Out: MallocSizeOf,
 	T: MallocSizeOf,
 	KF: KeyFunction<H>,
 	KF::Key: MallocSizeOf,
+	M: MallocSizeOfCallback<T>,
 {
 	fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
```

Comment:
It seems possible to call the old code when M::DO_RECURSIVE_SIZE_OF (a possible associated bool to M) is set to true, but is it worth it, I don't know (it only seems interesting for testing or benchmarking).

Reply:
not sure what you mean exactly, the current code is recursive as well as you can see from the test, the only difference is it is incremental

Reply:
Ah, I think now I get it, you mean in case there is a bug in the current impl, we could safely fall back to the old one? I expect the current code being faster simply because it is `O(1)` vs `O(n)`.

Reply:
yeah, I assume if one uses `Noop` impl, than `size_of` will not be called. The overhead introduced by `CountingCallback` is negligible (a couple usize increments). So, yes, not really worth it.